### PR TITLE
Fixed errors

### PR DESCRIPTION
**Pull request by sfreeman422**
- [x] Fixed defensive handling for empty/missing API responses in fun-fact.job.ts
- [x] Added Jest unit tests covering edge-case paths in fun-fact.job.spec.ts
- [x] Removed misleading `.catch(...)` wrapper in index.ts
- [x] Fixed `@typescript-eslint/no-explicit-any` lint error (replaced `any` with `Record<string, unknown>`)
- [x] Fixed Prettier formatting issue in fun-fact.job.spec.ts

### CODE REVIEW

File: packages/backend/src/jobs/fun-fact.job.ts [hunk 1]
```diff
@@ -150,18 +156,24 @@ export class FunFactJob {
...
-    const selected = response.data.selected[0];
-    const page = selected.pages[0];
+    const selected = response.data.selected?.[0];
+    if (!selected) {
+      throw new Error('Wikipedia OnThisDay API returned no "selected" events');
+    }
+    const page = selected.pages?.[0];
+    if (!page) {
+      throw new Error('Wikipedia OnThisDay API returned no pages for the selected event');
+    }
```
[the remark this answers is not in the document]
Comment (sfreeman422) [reply packages/backend/src/jobs/fun-fact.job.ts:176]: @copilot apply changes based on [this feedback](https://github.com/dev-chat/mocker/pull/228#discussion_r3009783604)

**Comment by sfreeman422** [comment]
@copilot failing checks
